do-not-commit/generate-injection.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python3
+import logging
+import os
+import subprocess
+import sys
+
+import yaml
+from jinja2 import Template
+
+logger = logging.getLogger(__file__)
+logger.setLevel(logging.INFO)
+handler = logging.StreamHandler(stream=sys.stderr)
+handler.setFormatter(
+    logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+)
+logger.addHandler(handler)
+
+if __name__ == "__main__":
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("template_file", type=str, help="The template file to use.")
+    parser.add_argument("config_file", type=str)
+
+    args = parser.parse_args()
+
+    with open(args.config_file, "r") as fin:
+        config = yaml.load(fin)
```

Comment:
Avoid using `load()`. `PyYAML.load` can create arbitrary Python
objects. A malicious actor could exploit this to run arbitrary
code. Use `safe_load()` instead. ([serialization.avoid-pyyaml-load.avoid-pyyaml-load](https://semgrep.dev/r/serialization.avoid-pyyaml-load.avoid-pyyaml-load))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=serialization.avoid-pyyaml-load.avoid-pyyaml-load) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>